solver/src/solver.rs

```diff
@@ -15,6 +15,9 @@ use reqwest::Url;
 use shared::{price_estimate::PriceEstimating, token_info::TokenInfoFetching};
 use structopt::clap::arg_enum;
 
+/// Defines shared behaviour for all Solvers.
+/// A `solve` method which transforms a collection of `Liquidity` sources into a `Settlement`
```

Comment:
into a list of Settlement ?

Generally I don't find this comment much helpful as it's just stating the obvious that can be seen from the method definition (and this sentence will likely forgotten to be updated if the method definition changes). Maybe we could explain a bit more what "Solvers" are how (e.g. that they can use liquidity on top of the provided ones if they have private sources, etc.)